x-pack/qa/security-example-authorization-engine/build.gradle

```diff
@@ -0,0 +1,46 @@
+apply plugin: 'elasticsearch.esplugin'
```

Comment:
I think we need to answer two questions:

1. Are we ok having more dependencies on x-pack code outside of the x-pack directory?
2. Are we ok making this code ASL 2.0 licensed?

For 2, I do not see an issue since the previous https://github.com/elastic/shield-custom-realm-example/blob/master/LICENSE.txt[example] was ASL 2.0 licensed. I think there may be more concern about 1; @rjernst @jasontedor any thoughts.

Reply:
I don't think depending on/using xpack outside the xpack directory is a problem, when the code in question will be ASL 2.0 licensed anyways. I think that is better from a visibility standpoint, and also clearer, since having ASL code under the x-pack dir could be easily overlooked. So putting this example (and probably moving the existing xpack examples) to plugins/examples makes sense to me.

Reply:
@atorok @rjernst I moved the example added here; can one of you take a look at that part?

I'll open up a separate change for the existing example.

Reply:
thanks @jaymode looks good.